src/graph/optimizer/rule/UnionAllIndexScanBaseRule.cpp

```diff
@@ -62,20 +96,77 @@ StatusOr<TransformResult> UnionAllIndexScanBaseRule::transform(OptContext* ctx,
   auto scan = static_cast<const IndexScan*>(node);
 
   auto metaClient = ctx->qctx()->getMetaClient();
-  StatusOr<std::vector<std::shared_ptr<meta::cpp2::IndexItem>>> status;
-  if (node->kind() == graph::PlanNode::Kind::kTagIndexFullScan) {
-    status = metaClient->getTagIndexesFromCache(scan->space());
-  } else {
-    status = metaClient->getEdgeIndexesFromCache(scan->space());
-  }
+  auto status = node->kind() == graph::PlanNode::Kind::kTagIndexFullScan
+                    ? metaClient->getTagIndexesFromCache(scan->space())
```

Comment:
What's about getTagIndexByNameFromCache(spaceId, tagName) ?

Reply:
Did you mean `getTagIndexByNameFromCache()`? Sounds like something we could optimize. 
The current approach is:
1. get all indexItems in the space
2. check if indexItems is empty
3. `OptimizerUtils::findOptimalIndex(operand, indexItems, &isPrefixScan, &ictx)`

which can be turned into:
1. analyze the expression to get tag info
2. get indexItems using`getTagIndexByNameFromCache(spaceID, tagName)`
3. check if indexItems is empty
4. `OptimizerUtils::findOptimalIndex(operand, indexItems, &isPrefixScan, &ictx)`

I'll mark this as a TODO. WDYT?